test_conformance/subgroups/subhelpers.h

```diff
@@ -1080,7 +1080,7 @@ template <typename Ty>
 typename std::enable_if<TypeManager<Ty>::is_sb_scalar_type::value>::type
 set_value(Ty &lhs, const cl_ulong &rhs)
 {
-    lhs.data = rhs;
+    lhs.data = cl_half_from_float(static_cast<cl_float>(rhs), CL_HALF_RTE);
```

Comment:
I think this should also query the rounding mode of the device to select between `CL_HALF_R*` to ensure the correct rounding behaviour.

Reply:
Modified to detect and use the device's rounding mode

Reply:
The query used for image tests isn't going to work correctly since it's specific to image writes and unrelated to the default rounding mode (used by the arithmetic operators and conversions). Also, it would execute an image write kernel for every data point in the test (!).

The default mode is always RTE for the full profile, so I think @Jeremy-Kemp  might be referring to the embedded profile, where it can be either RTE or RTZ. You can check this with a CL_DEVICE_SINGLE_FP_CONFIG query, which would be faster than running a kernel, but we'd want to check the impact on run-time before adding it to an arithmetic operator implementation.

The only alternative I can see would be to provide another argument to set_value to pass in the rounding mode. This would be needed anyway if we ever want to use this to support explicit conversion operators with different rounding modes.

Reply:
Agreed and modifying to use  CL_DEVICE_SINGLE_FP_CONFIG query and to allow RTZ for embedded.
This wouldn't have run the kernel for every data point though, coz the detection code is called only once and then used to initialize a global.

Reply:
> This wouldn't have run the kernel for every data point though, coz the detection code is called only once and then used to initialize a global.

Ha, of course, another brain cramp..